internal/pkg/server/server.go

```diff
@@ -38,8 +39,20 @@ func NewServer(opt ServerOpts) *Server {
 	g.GET("/", func(c echo.Context) error {
 		return c.HTML(http.StatusOK, indexHTML)
 	})
-	g.POST("/tar", s.postTar)
-	g.GET("/tar", s.getTar)
+	g.POST("/tar", func(c echo.Context) error {
+		err := s.postTar(c)
+		if err != nil {
+			return c.HTML(errorHTTPMapping(err), errorHTML)
```

Comment:
what do you think about implementing the error mapping like this? it would allow for error handling in middlewares and showing the original error message if enabling echo.Debug.

```suggestion
			return dockerToEchoErrorMapping(err)
```

```go
func dockerToEchoErrorMapping(err error) *echo.HTTPError {
	switch {
	case strings.Contains(err.Error(), "forbidden"):
		return echo.NewHTTPError(http.StatusForbidden, err.Error())
	case strings.Contains(err.Error(), "not found"):
		return echo.NewHTTPError(http.StatusNotFound, err.Error())
...
	}
	return echo.NewHTTPError(http.StatusInternalServerError, err.Error())
}
```